app/src/lib/stores/app-store.ts

```diff
@@ -5297,6 +5304,26 @@ export class AppStore extends TypedBaseStore<IAppState> {
             FetchType.UserInitiatedTask
           )
         }
+      } else if (isRefInUpstream) {
+        // the remote ref will be something like `upstream/my-cool-branch`
+        const remoteRef = `${UpstreamRemoteName}/${head.ref}`
+
+        const remoteRefExists =
+          gitStore.allBranches.find(branch => branch.name === remoteRef) != null
+
+        // only try a fetch here if we can't find the ref
+        if (!remoteRefExists) {
+          const remotes = await getRemotes(repository)
+          const remoteUpstream = findUpstreamRemote(gitHubRepository, remotes)
+
+          if (remoteUpstream) {
+            await this._fetchRemote(
+              repository,
+              remoteUpstream,
+              FetchType.UserInitiatedTask
+            )
+          }
+        }
```

Comment:
i have some questions about this flow, but it looks like @rafeca and @tierninho have already been discussing this. can you share the current understanding here? it sounds like the fork case is broken here, but also in production? is that right?

would it make sense to file a separate bug for this problem and close it with  #9382? (I'm having a little trouble following here distinguishing the bug and which environments its present in, and therefore whether it's a regression or not)

Reply:
I made this change in 0a33f905d3ed692c28537bf153eea22fc3fb973c to fix the issue described in https://github.com/desktop/desktop/pull/9340#issuecomment-603935866.

Since the issue that it fixes is not important (and is reproducible on production) I'm gonna revert this commit from this PR and move it to https://github.com/desktop/desktop/pull/9382, where it makes much more sense. This will also make this PR a bit smaller and focused on actually displaying PRs from upstream repos.

👍to create an issue for https://github.com/desktop/desktop/pull/9382, since I just realized that this whole thing got a bit out of control 😅

Reply:
awesome, thank you @rafeca!